src/filesystem.cpp

```diff
@@ -685,194 +670,88 @@ static std::optional<std::string> get_games_path()
-		DBG_FS << "userdata is relative to the current working directory: " << dir.string();
-	}
-
-	return dir;
-}
-#else
-static bfs::path linux_userdata(const std::string& newprefdir)
-{
-	bfs::path dir;
-	std::string temp = newprefdir;
-	const char* home_str = getenv("HOME");
-	char const* xdg_data = getenv("XDG_DATA_HOME");
-
-#ifdef PREFERENCES_DIR
-	if(temp.empty()) {
-		temp = PREFERENCES_DIR;
-		DBG_FS << "userdata using PREFERENCES_DIR '" << PREFERENCES_DIR << "'";
-		if(home_str && temp[0] == '~') {
-			temp = home_str + temp.substr(1);
-			DBG_FS << "userdata is relative to HOME";
+		const char* h = std::getenv("HOME");
+		std::string home = h ? h : "";
+		if(!home.empty()) {
+			dir = home;
+		} else {
+			dir = get_cwd();
+			ERR_FS << "Unable to determine path to user's HOME.";
+			WRN_FS << "Using current directory instead: " << dir.string();
 		}
-	}
 #endif
-
-	// if a custom userdata dir is not specified
-	// and one of the XDG home and regular home env variables are not empty
-	// then use one of those
-	if(temp.empty() && ((xdg_data && xdg_data[0] != '\0') || home_str)) {
-		if(xdg_data && xdg_data[0] != '\0') {
-			dir = xdg_data;
-			DBG_FS << "userdata using XDG_DATA_HOME";
-		} else if(home_str) {
-			dir = home_str;
-			dir /= ".local/share";
-			DBG_FS << "userdata using HOME";
-		}
-
-		dir /= "wesnoth";
-		dir /= get_version_path_suffix();
-		return dir;
-	}
-
-	// if a custom userdata dir using an absolute path was specified, just use that
-	if(!temp.empty() && temp[0] == '/') {
-		dir = temp;
-		DBG_FS << "userdata is an absolute path";
-		return dir;
-	}
-
-	// if no custom userdata folder was specified and there's also no XDG/HOME variables available, set a default folder name
-	if(temp.empty()) {
-		temp = ".wesnoth" + get_version_path_suffix();
+		dir /= newprefdir.substr(1);
+	} else {
+		dir = newprefdir;
 	}
-
-	// unable to determine another userdata directory, so just use the current working directory for the userdata
-	dir = get_cwd();
-	dir /= temp;
-	ERR_FS << "unable to determine location to use for userdata, defaulting to current working directory: " << dir.string();
-
-	return dir;
-}
-#endif
-
-void set_user_data_dir(std::string newprefdir)
-{
-#ifdef _WIN32
-	DBG_FS << "determining Windows userdata dir";
-	user_data_dir = windows_userdata(newprefdir);
-#elif defined(__APPLE__) || defined(WESNOTH_BOOST_OS_IOS)
-	DBG_FS << "determining macOS userdata dir";
-	user_data_dir = apple_userdata(newprefdir);
-#else
-	DBG_FS << "determining non-Windows and non-macOS userdata dir";
-	user_data_dir = linux_userdata(newprefdir);
-#endif
+	user_data_dir = dir;
+	DBG_FS << "userdata dir set to: " << user_data_dir.string();
 
 	setup_user_data_dir();
-	user_data_dir = normalize_path(user_data_dir.string(), true, true);
+	// normalize_path expects the path to exist so calling it after potentially creating it in setup_user_data_dir
+	dir = normalize_path(user_data_dir.string(), true, true);
+	if (!dir.empty()) {
```

Comment:
Should be no space between the `if` and `(`.

Reply:
Fixed.